src/object-store/src/error.rs

```diff
@@ -0,0 +1,45 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::any::Any;
+
+use common_error::ext::ErrorExt;
+use common_error::status_code::StatusCode;
+use common_macro::stack_trace_debug;
+use snafu::{Location, Snafu};
+
+#[derive(Snafu)]
+#[snafu(visibility(pub))]
+#[stack_trace_debug]
+pub enum Error {
+    #[snafu(display("Global storage not found: {}", global_object_store))]
+    GlobalStorageNotFound {
+        location: Location,
+        global_object_store: String,
+    },
+}
+
+pub type Result<T> = std::result::Result<T, Error>;
+
+impl ErrorExt for Error {
+    fn status_code(&self) -> StatusCode {
+        match self {
+            Error::GlobalStorageNotFound { .. } => StatusCode::StorageUnavailable,
```

Comment:
Is this an `InvalidArguments`? Since we are looking up a storage which doesn't exist.

Reply:
> Since we are looking up a storage which doesn't exist.

We don't need to have such an error because the `ObjectStoreManager::find` method returns `Option<ObjectStore>`.

Reply:
This error is returned by `try_new`, `StorageUnavailable` isn't a proper code. This is caused by passing incorrect `default_object_store` or `stores` map to `try_new()`, which is a kind of invalid argument.

`StorageUnavailable` usually means the storage engine is temporarily unavailable ("service unavailable"). e.g. Timeout while reading/writing object store. Not for representing an "object store doesn't exist".
https://github.com/GreptimeTeam/greptimedb/blob/1658d088abbc232604fb1caeafe649c9f1358148/src/common/error/src/status_code.rs#L65-L66

Reply:
Sorry for my miscommunication because I originally thought your point is relating to `GlobalStorageNotFound`, not `StatusCode::StorageUnavailable`. Thank you for the polite response.